### PR TITLE
Provide descriptions

### PR DESCRIPTION
**Pull request by deads2k**
Add human readable descriptions for scopes.

@sgallagher last commit only.

### CODE REVIEW

File: pkg/authorization/authorizer/scope/converter.go [hunk 1]
```diff
@@ -95,6 +96,17 @@ func (userEvaluator) Validate(scope string) error {
 	return fmt.Errorf("unrecognized scope: %v", scope)
 }
 
+func (userEvaluator) Describe(scope string) string {
+	switch scope {
+	case UserIndicator + UserInfo:
+		return "information about you, including username, identity names, and group membership"
```
Comment (sgallagher) [inline pkg/authorization/authorizer/scope/converter.go]: If this information is going to be returned to a user, should it be properly capitalized and punctuated?

Also, grammar-nit; needs a colon (but I agree on the use of the Oxford Comma):
"Information about you, including: username, identity names, and group membership."

File: pkg/authorization/authorizer/scope/converter.go [hunk 2]
```diff
@@ -95,6 +96,17 @@ func (userEvaluator) Validate(scope string) error {
...
+	case UserIndicator + UserAccessCheck:
+		return `information about what you can do: "can I create builds"`
```
Comment (sgallagher) [inline pkg/authorization/authorizer/scope/converter.go]: Suggest:
Information about user privileges, e.g. "Can I create builds?"

File: pkg/authorization/authorizer/scope/converter.go [hunk 3]
```diff
@@ -118,36 +130,46 @@ func (clusterRoleEvaluator) Handles(scope string) bool {
 }
 
 func (e clusterRoleEvaluator) Validate(scope string) error {
+	_, _, err := e.getRoleNamespace(scope)
+	return err
+}
```
Comment (sgallagher) [inline pkg/authorization/authorizer/scope/converter.go]: This belongs in the patch that introduces the `Validate()` method.

File: pkg/authorization/authorizer/scope/converter.go (hunk 1 above)
Comment (deads2k) [inline pkg/authorization/authorizer/scope/converter.go]: > If this information is going to be returned to a user, should it be properly capitalized and punctuated?
> 
> Also, grammar-nit; needs a colon (but I agree on the use of the Oxford Comma):
> "Information about you, including: username, identity names, and group membership."

@jwforres @fabianofranz I think that @liggitt's intent is for this to be a human readable string you can present to a user to describe a scope.  I don't know precisely how he intends for you to get the information in the console, but in the CLI this is pretty easy to get.  Should it be a complete sentence?

**Comment by sgallagher** [comment]
Modulo the specific string contents under discussion, LGTM.

**Comment by deads2k** [comment]
> Modulo the specific string contents under discussion, LGTM.

Updated strings.  We can always change our mind later.  

[merge]

**Pull request merged**